crates/task-impls/src/consensus.rs

```diff
@@ -1295,52 +1307,37 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
     }
 }
 
-impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I>> TS
+impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> + 'static> TaskState
     for ConsensusTaskState<TYPES, I, A>
 {
-}
-
-/// Type alias for Consensus task
-pub type ConsensusTaskTypes<TYPES, I, A> = HSTWithEvent<
-    ConsensusTaskError,
-    HotShotEvent<TYPES>,
-    ChannelStream<HotShotEvent<TYPES>>,
-    ConsensusTaskState<TYPES, I, A>,
->;
-
-/// Event handle for consensus
-pub async fn sequencing_consensus_handle<
-    TYPES: NodeType,
-    I: NodeImplementation<TYPES>,
-    A: ConsensusApi<TYPES, I> + 'static,
->(
-    event: HotShotEvent<TYPES>,
-    mut state: ConsensusTaskState<TYPES, I, A>,
-) -> (
-    std::option::Option<HotShotTaskCompleted>,
-    ConsensusTaskState<TYPES, I, A>,
-) {
-    if let HotShotEvent::Shutdown = event {
-        (Some(HotShotTaskCompleted::ShutDown), state)
-    } else {
-        state.handle_event(event).await;
-        (None, state)
+    type Event = HotShotEvent<TYPES>;
+    type Output = ();
+    fn filter(&self, event: &HotShotEvent<TYPES>) -> bool {
+        !matches!(
+            event,
+            HotShotEvent::QuorumProposalRecv(_, _)
+                | HotShotEvent::QuorumVoteRecv(_)
+                | HotShotEvent::QCFormed(_)
+                | HotShotEvent::DACRecv(_)
+                | HotShotEvent::ViewChange(_)
+                | HotShotEvent::SendPayloadCommitmentAndMetadata(..)
+                | HotShotEvent::Timeout(_)
+                | HotShotEvent::TimeoutVoteRecv(_)
+                | HotShotEvent::VidDisperseRecv(..)
+                | HotShotEvent::Shutdown,
+        )
```

Comment:
Is it ok if I make an issue and come back to this since it's the same as before?  I'm sort of ok with this pattern because it's opt in so when new events are added no tasks will receive so the default is no functional change.  It does mean you have to update the filter if you care about handling the event you added.  Which I could see being easy to forget.

Mostly I'm tired of resolving merge conflicts and this is will be a bunch more busy work I'd rather do in another branch